src/Functions/array/arrayDistance.cpp

```diff
@@ -333,19 +377,59 @@ class FunctionArrayDistance : public IFunction
             typename Kernel::template State<Float64> state;
             for (size_t i = 0; prev < off; ++i, ++prev)
             {
-                Kernel::template accumulate<Float64>(state, data_x[i], data_y[prev]);
+                Kernel::template accumulate<Float64>(state, data_x[i], data_y[prev], kernel_params);
             }
-            result_data[row] = Kernel::finalize(state);
+            result_data[row] = Kernel::finalize(state, kernel_params);
             row++;
         }
         return result;
     }
 
+    typename Kernel::ConstParams initConstParams(const ColumnsWithTypeAndName &) const { return {}; }
 };
 
+
+template <>
+size_t FunctionArrayDistance<LpDistance>::getNumberOfArguments() const { return 3; }
+
+template <>
+ColumnNumbers FunctionArrayDistance<LpDistance>::getArgumentsThatAreAlwaysConstant() const { return {2}; }
+
+template <>
+LpDistance::ConstParams FunctionArrayDistance<LpDistance>::initConstParams(const ColumnsWithTypeAndName & arguments) const
+{
+    if (arguments.size() < 3)
+        throw Exception(
+                    ErrorCodes::LOGICAL_ERROR,
+                    "Argument p of function {} was not provided",
+                    getName());
+
+    if (!arguments[2].column->isNumeric())
+        throw Exception(
+                    ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT,
+                    "Argument p of function {} must be numeric constant",
+                    getName());
+
+    if (!isColumnConst(*arguments[2].column) && arguments[2].column->size() != 1)
+        throw Exception(
+                    ErrorCodes::ILLEGAL_COLUMN,
+                    "Second argument for function {} must be either constant Float64 or constant UInt",
+                    getName());
+
+    Float64 p = arguments[2].column->getFloat64(0);
+    if (p < 1 || p == HUGE_VAL)
```

Comment:
Why `p == HUGE_VAL`, but not `p >= HUGE_VAL` ?

Reply:
changed to >=